Sources/Clappr_iOS/Classes/Plugin/Core/MediaControl/PlayButton.swift

```diff
@@ -64,27 +80,40 @@ open class PlayButton: MediaControl.Element {
         view.isHidden = false
     }
 
-    @objc func togglePlayPause() {
-        guard let playback = activePlayback else {
-            return
-        }
+    @objc open func togglePlayPause() {
+        guard let playback = activePlayback else { return }
 
-        if playback.state == .playing {
-            activePlayback?.pause()
-        } else if playback.state == .paused {
-            activePlayback?.play()
+        switch playback.state {
+        case .playing:
+            pause()
+        case .paused, .idle:
+            play()
+        default:
+            break
         }
     }
 
-    public func changeIcon() {
-        guard let playback = activePlayback else {
-            return
-        }
-
-        if playback.state == .paused {
-            button.setImage(playIcon, for: .normal)
-        } else if playback.state == .playing {
-            button.setImage(pauseIcon, for: .normal)
-        }
+    private func pause() {
+        guard let playback = activePlayback, playback.canPause else { return }
```

Comment:
Once we are moving `play` and `pause` logic to `Container` in order to improve it's encapsulating it would be interesting to evaluate if this statement should be also there.